src/tasks/AndroidAppBuilder/Templates/monodroid-coreclr.c

```diff
@@ -130,9 +129,64 @@ get_tpas_from_path(const char* dir_path, const char** tpas)
 }
 
 static int
-mono_droid_runtime_init (const char* executable, int managed_argc, char* managed_argv[], int local_date_time_offset)
+bundle_executable_path (const char* executable, const char* bundle_path, const char** executable_path)
 {
-    LOG_INFO ("mono_droid_runtime_init called with executable: %s", executable);
+    size_t executable_path_len = strlen(bundle_path) + strlen(executable) + 1; // +1 for '/'
+    char* temp_path = (char*)malloc(sizeof(char) * executable_path_len + 1); // +1 for '\0'
+    if (temp_path == NULL)
+    {
+        return -1;
+    }
+
+    size_t res = snprintf(temp_path, (executable_path_len + 1), "%s/%s", bundle_path, executable);
+    if (res < 0 || res != executable_path_len)
+    {
+        return -1;
+    }
+    *executable_path = temp_path;
+    return executable_path_len;
+}
+
+static void
+free_resources ()
+{
+    if (g_bundle_path)
+    {
+        free (g_bundle_path);
+        g_bundle_path = NULL;
+    }
+    if (g_executable_path)
+    {
+        free (g_executable_path);
+        g_executable_path = NULL;
+    }
+    if (g_coreclr_handle)
+    {
+        coreclr_shutdown (g_coreclr_handle, g_coreclr_domainId);
+        g_coreclr_handle = NULL;
+    }
+}
+
+static int 
+mono_droid_execute_assembly (const char* executable_path, void* coreclr_handle, unsigned int coreclr_domainId, int managed_argc, const char** managed_argv)
+{
+    unsigned int rv;
+    LOG_INFO ("Calling coreclr_execute_assembly");
+    coreclr_execute_assembly (coreclr_handle, coreclr_domainId, managed_argc, managed_argv, executable_path, &rv);
+    LOG_INFO ("Exit code: %d.", rv);
+    return rv;
+}
+
+static int
+mono_droid_runtime_init (const char* bundle_path, const char* executable, int local_date_time_offset)
+{
+    LOG_INFO ("mono_droid_runtime_init (CoreCLR) called with executable: %s", executable);
```

Comment:
Is the `bundle_path` important here?

Reply:
Not important, we can use the global `g_bundle_path` instead.